### PR TITLE
Adds tutorial for working with function, computer, hybrid, and inplac…

### PR DESCRIPTION
**Pull request by zjpetersen**
…e ops

### CODE REVIEW

**Comment by zjpetersen** [comment]
@etarena Could you take a look at this tutorial for using different types of ops (function, computer, hybrid, and inplace).  Mostly I'm looking for critiques about areas that could be more clear to people who are just starting off with Ops.

**Comment by etarena** [comment]
@zjpetersen no worries dude. i'll go over it first thing tomorrow morning - eh?

File: using-ops/src/main/java/UsingOpTypes.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		long start = System.currentTimeMillis();
+		for (int i = 0; i < 1000; i++) {			
+			ij.op().math().add(out, in, 5);  //Searches for the op everytime and then runs it
+		}
+		long mid = System.currentTimeMillis();
+		for (int i = 0; i < 1000; i++) {
+			add5.compute(in, out);	//Just runs the op
+		}
+		long end = System.currentTimeMillis();
+		ij.log().info("--------- Slow = " + (mid - start) + "ms ---------");
```
Comment (imagejan) [inline using-ops/src/main/java/UsingOpTypes.java]: Speaking about performance, you could save about 20 calls to `ij.log()` if you did `log = ij.log()` and then simply `log.info("...")`, no?

**Comment by zjpetersen** [comment]
Good ideas @etarena and @imagejan.  I'll make those updates.

**Comment by ctrueden** [comment]
@imagejan There is almost certainly no performance gain to caching the local variable. The JIT is very smart about that stuff now. So @zjpetersen, I wouldn't worry about it.

**Comment by ctrueden** [comment]
OK, I take it back. In my benchmarks, `ij.log()` is indeed slower, based on this code:
```java
import net.imagej.ImageJ;

import org.scijava.log.LogService;

public class Sandbox {

	public static void main(final String... args) throws Exception {
		ImageJ ij = new ImageJ();
		final LogService[] logs = new LogService[10000];
		final long start = System.currentTimeMillis();
		for (int i=0; i<logs.length; i++) {
			logs[i] = ij.log();
		}
		final long end = System.currentTimeMillis();
		System.out.println("Elapsed = " + (end - start) + " ms");
	}
}
```

Changing `ij.log()` to just `log` which was cached outside the loop makes the output on my system go from ~25ms to 0ms.

**Comment by zjpetersen** [comment]
I included the change by @imagejan but @ctrueden is right in that it's not really related to the point of this tutorial either way.

**Comment by stelfrich** [comment]
I felt free to add the missing pieces to the special ops tutorial and squashed my changes into the WIP commit (effectively making it non-WIP 😄 ).

Could someone please take a second look at it? I really think we should get that tutorial merged asap..

**Comment by ctrueden** [comment]
@stelfrich I really appreciate your efforts, and just wanted to let you know I haven't forgotten about this. It is one of 70 mails in my backlog. I will get to it as soon as I can. 😄

**Comment by ctrueden** [comment]
@stelfrich Bug me about this during the hackathon?

**Comment by stelfrich** [comment]
Will do @ctrueden! Maybe when @imagejan is around as well..

**Comment by stelfrich** [comment]
@ctrueden Hmm, so much about bugging you at the hackathon :pensive: Sorry!

**Comment by ctrueden** [comment]
@stelfrich No worries; this is all going to change—_again_—with my Nil updates.

**Comment by ctrueden** [comment]
Rebased over the latest master. Gonna merge it, even though it is still not done, because it is useful and working.

**Pull request merged**